pkg/operator/serviceaccountissuercontroller/serviceaccountissuer_controller.go

```diff
@@ -60,31 +73,27 @@ func (c *ServiceAccountIssuerController) sync(ctx context.Context, controllerCon
 
 	// there is no service account issuer set and there are no service account issuers in status, no-op.
 	if len(currentIssuer) == 0 && len(operator.Status.ServiceAccountIssuers) == 0 {
-		return nil
-	}
-
-	// if the status is not set, but the current issuer is not empty, update the status and let the controller resync
-	if len(currentIssuer) > 0 && len(operator.Status.ServiceAccountIssuers) == 0 {
 		operatorCopy := operator.DeepCopy()
-		operatorCopy.Status.ServiceAccountIssuers = []operatorv1.ServiceAccountIssuerStatus{
-			{
-				Name: currentIssuer,
-			},
-		}
+		operatorCopy.Status.ServiceAccountIssuers = defaultServiceAccountIssuerValue
 		_, statusUpdateErr := c.kubeAPIServerOperatorClient.UpdateStatus(ctx, operatorCopy, metav1.UpdateOptions{})
 		if statusUpdateErr == nil {
-			klog.Infof("ServiceAccountIssuer %q is now set as active issuer", currentIssuer)
+			klog.Infof("ServiceAccountIssuer %q is set as a default value", defaultServiceAccountIssuerValue[0].Name)
 			statusUpdateErr = factory.SyntheticRequeueError
 		}
 		return statusUpdateErr
 	}
 
-	// in this case, user deleted service account issuer value, which means we have to remove all values from the status
-	// if we don't do this and instead just remove first value, the new tokens will be issued with "previous" service
-	// account issuer which is not desired
-	if len(currentIssuer) == 0 && len(operator.Status.ServiceAccountIssuers) != 0 {
+	// this is a case when issuer is not set in auth config and the operator status already has the default issuer set.
+	if isDefaultServiceAccountIssuer(currentIssuer, operator.Status.ServiceAccountIssuers) {
```

Comment:
why don't we instead set `currentIssuer` to `https://kubernetes.default.svc` if `len(currentIssuer) == 0`?

Reply:
because that will need a write to authConfig and I don't want to write default into auth config spec from a controller :-)

Reply:
I meant (line 67)
```go
currentIssuer := authConfig.Spec.ServiceAccountIssuer
if len(currentIssuer) == 0 {
    currentIssuer = "https://kubernetes.default.svc"
}

```

Reply:
@stlaz it will complicate the the revocation case and when I do this,random unit test start to fail (like pruning :-)